File: string/112.py
def applyBackspace(s: str):
    stack = []

    for ch in s:
        if ch.isalpha():
            stack.append(ch)
        elif ch == "#" and len(stack) > 0:
            stack.pop()

    return "".join(stack)


def backspaceCompare(s: str, t: str) -> bool:
    return applyBackspace(s) == applyBackspace(t)


print(backspaceCompare("a#c", "b"))

File: string/112.cpp
#include <bits/stdc++.h>

using namespace std;

string applyBackspace(string cadena) {
    stack<char> op;

    for(char const & ch: cadena) {
        if(isalpha(ch))
            op.push(ch);
        else if(ch == '#' && !op.empty())
            op.pop();
    }

    string temp = "";
    while(!op.empty()) {
        temp += op.top();
        op.pop();
    }

    reverse(temp.begin(), temp.end());
    return temp;
}

bool backspaceCompare(string s, string t) {
    return applyBackspace(s) == applyBackspace(t);
}

int main() {
    string s = "a#c", t = "b";
    cout<<boolalpha;
    cout<<backspaceCompare(s, t);

    return 0;
}